R/tag_examplesShinylive.R

```diff
@@ -0,0 +1,188 @@
+#'
+#' # A typical example would be:
+#' #' (docs)
+#' #' @examplesShinylive
+#' #' library(<your package>)
+#' #' interactive <- function() TRUE
+#' #' {{ next_example }}
+#' #' @examples
+#' #' app <- ...
+#' #' if (interactive()) {
+#' #'   shinyApp(app$ui, app$server)
+#' #' }
+#'
+#' # multiple apps:
+#' #' (docs)
+#' #' @examplesShinylive
+#' #' @examples
+#' #' (your example app 1)
+#' #' @examplesShinylive
+#' #' @examples
+#' #' (your example app 2)
+#'
+#' # skip parts of example code:
+#' #' (docs)
+#' #' @examples
+#' #' (your example code - skipped)
+#' #' @examplesShinylive
+#' #' @examples
+#' #' (your example code - included)
+#'
+#' # multiple apps with keywords:
+#' #' (docs)
+#' #' @examplesShinylive
+#' #' x <- 1
+#' #' {{ next_example }}
+#' #' @examples
+#' #' (your example app 1)
+#' #' @examplesShinylive
+#' #' y <- 1
+#' #' {{ next_example }}
+#' #' @examples
+#' #' (your example app 2)
+NULL
+
+#' @noRd
+#' @exportS3Method roxygen2::roxy_tag_parse roxy_tag_examplesShinylive
+#' @importFrom glue glue_data
+#' @importFrom stringr str_trim
+#' @importFrom roxygen2 warn_roxy_tag
+roxy_tag_parse.roxy_tag_examplesShinylive <- function(x) {
+  if (stringr::str_trim(x$raw) == "") {
+    x$raw <- "{{ next_example }}"
+  }
+
+  # not elegant but this is the most efficient way to access sibling tags
+  tokens <- get("tokens", envir = parent.frame(3L))
+
+  tags_examples <- Filter(function(x) x$tag %in% c("examples", "examplesIf"), tokens)
+
+  examples <- lapply(tags_examples, `[[`, "raw")
+
+  next_example <- Reduce(
+    function(x, y) `if`(x$line < y$line, x, y),
+    Filter(function(y) y$line > x$line, tags_examples)
+  )$raw
+
+  prev_example <- Reduce(
+    function(x, y) `if`(x$line > y$line, x, y),
+    Filter(function(y) y$line < x$line, tags_examples)
+  )$raw
+
+  x$raw <- try(
+    as.character(
+      glue::glue_data(
+        .x = list(
+          tags_examples = tags_examples,
+          examples = examples,
+          next_example = next_example,
+          prev_example = prev_example
+        ),
+        x$raw,
+        .open = "{{",
+        .close = "}}"
+      )
+    ),
+    silent = TRUE
+  )
+
+  if (inherits(x$raw, "try-error")) {
+    roxygen2::warn_roxy_tag(x, "failed to interpolate the content")
+    return(NULL)
+  }
+
+  if (is.null(x$raw) || length(x$raw) == 0) {
+    roxygen2::warn_roxy_tag(x, "requires a value")
+    return(NULL)
+  }
+  x$val <- create_shinylive_url(x$raw) # nolint: object_usage_linter.
+  x
+}
+
+#' @noRd
+#' @exportS3Method roxygen2::roxy_tag_rd roxy_tag_examplesShinylive
+#' @importFrom roxygen2 rd_section
+roxy_tag_rd.roxy_tag_examplesShinylive <- function(x, base_path, env) {
+  roxygen2::rd_section("examplesShinylive", x$val)
+}
+
+#' @noRd
+#' @exportS3Method format rd_section_examplesShinylive
+format.rd_section_examplesShinylive <- function(x, ...) {
+  iframe_attrs <- paste(
+    "height=\"800\"",
+    "width=\"150\\%\"", # @TODO: find a better way to set the width
```

Comment:
can't find a good value

150% looks like this:
(we have overly of menu on right and the app)
![image](https://github.com/user-attachments/assets/f3071bd0-e434-4d9a-8ed2-d6f6af3dafc0)

100% looks like this:
(teal app auto-collapse)
![image](https://github.com/user-attachments/assets/2b81bd31-56aa-487e-bbb7-6744aa367f4b)

Reply:
OK I think I found a sweet spot - is to fit to 100% width and apply the 80% scaling
At the end it looks like this:
<img width="1210" alt="image" src="https://github.com/user-attachments/assets/0ce7446f-3e4b-4cd2-a29d-56c3ceef3a85">